src/components/SummaryKeyComponents/SummaryKeyComponent.js

```diff
@@ -0,0 +1,113 @@
+import { getPaperTldrAndAbstract } from "../../services/SemanticScholarService";
+import { TLDRAccordionItem } from "./TLDRAccordionItem";
+import { AbstractAccordionItem } from "./AbstractAccordionItem";
+import { AbstractSummaryAccordionItem } from "./AbstractSummaryAccordionItem";
+import { SelectionSummaryAccordionItem } from "./SelectionSummaryAccordionItem";
+import {
+  EventHandlerService,
+  PDFLEvents,
+} from "../../services/EventHandlerService";
+
+/**
+ * Component responsible for displaying the sidebar for summaries/key
+ * @property {Object} components object that holds DOM elements that represent this component, as well as component's context
+ * @property {HTMLElement} components.accordionItem accordion item
+ * @property {HTMLElement} components.closeBtn button that closes sidepage
+ * @property {TLDRAccordionItem} tldrItem the TLDR Accordion Item instance
+ * @property {AbstractAccordionItem} abstractItem the Abstract Accordion Item instance
+ * @property {AbstractSummaryAccordionItem} abstractSummaryItem the Abstract Summary Accordion Item instance
+ * @property {SelectionSummaryAccordionItem} selectionSummaryItem the Selection Summary Accordion Item instance
+ */
+class SummaryKeyComponent {
+  components = {
+    sidePageSummary: document.querySelector("#side-page-summary"),
+    accordionItem: document.getElementsByClassName("accordion"),
+    closeBtn: document.querySelector("#close-btn-summary"),
+  };
+
+  /**
+   * Creates and initializes new Summary Key Component
+   * @constructor
+   */
+  constructor() {
+    this.tldrItem = new TLDRAccordionItem();
+    this.abstractItem = new AbstractAccordionItem();
+    this.abstractSummaryItem = new AbstractSummaryAccordionItem();
+    this.selectionSummaryItem = new SelectionSummaryAccordionItem();
```

Comment:
These 4 could have been the same class instantiated 4 times.

Reply:
So, maybe give the dom id in the constructor and create just one class?
And at this point since clear, setText and setError are a little bit different from each other I move the logic in the SummaryKeyComponent?

Reply:
setText can just take text and set it, so from the outside you take care of how to prepare it, for setError you can pass the error message throught the constructor